querier.go

```diff
@@ -544,6 +544,7 @@ func (s *populatedChunkSeries) Next() bool {
 			chks = chks[1:]
 		}
 
+		validChks := make([]ChunkMeta, 0, len(chks))
```

Comment:
Careful! This means an allocation on every loop iteration which did not exist before.
Maybe we can get around this by inserting into the same slice again at a position `j <= i`, where `j` gets smaller by `1` for each skipped chunk?

Reply:
Sorry, did not even think about this while making the change. Will evaluate all my future changes about the potential performance impact.

Fixed it now.